source/tunneling/SecureTunnelingFeature.cpp

```diff
@@ -186,19 +183,19 @@ namespace Aws
                         return;
                     }
 
-                    string accessToken = response->ClientAccessToken->c_str();
-                    if (accessToken.empty())
+                    if (!response->ClientAccessToken.has_value() || response->ClientAccessToken->empty())
                     {
                         LOG_ERROR(TAG, "access token cannot be empty");
                         return;
                     }
+                    string accessToken = response->ClientAccessToken->c_str();
 
-                    string region = response->Region->c_str();
-                    if (region.empty())
+                    if (!response->Region.has_value() || response->Region->empty())
                     {
                         LOG_ERROR(TAG, "region cannot be empty");
                         return;
                     }
+                    string region = response->Region.value().c_str();
```

Comment:
very minor: consistency between this and line 191, even though they accomplish the same thing.